port/unix/omrsysinfo.c

```diff
@@ -7894,3 +7895,45 @@ omrsysinfo_get_processes(struct OMRPortLibrary *portLibrary, OMRProcessInfoCallb
 	return OMRPORT_ERROR_SYSINFO_NOT_SUPPORTED;
 #endif /* defined(AIXPPC) */
 }
+
+char *
+omrsysinfo_get_process_name(struct OMRPortLibrary *portLibrary, uintptr_t pid)
+{
+#if defined(AIXPPC)
+	int fd = 0;
+	char exebuf[35];
+	union {
+		struct psinfo info;
+		char raw[1024];
+	} infobuf;
+	snprintf(exebuf, sizeof(exebuf), "/proc/%" OMR_PRIuPTR "/psinfo", pid);
+	fd = open(exebuf, O_RDONLY);
+	if (-1 != fd) {
+		const char *name = infobuf.info.pr_fname;
+		ssize_t minsize = name - infobuf.raw + sizeof(infobuf.info.pr_fname);
+		ssize_t size = read(fd, infobuf.raw, sizeof(infobuf));
+		close(fd);
+		if (size >= minsize) {
+			uintptr_t len = strlen(name);
+			char *exename = portLibrary->mem_allocate_memory(portLibrary, len, OMR_GET_CALLSITE(), OMRMEM_CATEGORY_PORT_LIBRARY);
+			strcpy(exename, name);
+			return exename;
+		}
+	}
+#elif defined(LINUX) && !defined(OMRZTPF) /* defined(AIXPPC) */
+	ssize_t rc = 0;
+	char exebuf[32];
+	char linkbuf[PATH_MAX];
+	snprintf(exebuf, sizeof(exebuf), "/proc/%" OMR_PRIuPTR "/exe", pid);
+	rc = readlink(exebuf, linkbuf, sizeof(linkbuf));
+	if ((-1 != rc) && (rc < sizeof(linkbuf))) {
+		char *exename = portLibrary->mem_allocate_memory(portLibrary, rc + 1, OMR_GET_CALLSITE(), OMRMEM_CATEGORY_PORT_LIBRARY);
+		if (NULL != exename) {
+			strncpy(exename, linkbuf, rc);
+			exename[rc] = 0;
```

Comment:
Suggest `memcpy()` and clarify that we're storing a character:
```c
			memcpy(exename, linkbuf, rc);
			exename[rc] = '\0';
```
Maybe `exelen` (or similar) would be a better name than `rc`?